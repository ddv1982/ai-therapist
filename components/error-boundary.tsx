'use client';

import React, { Component, ReactNode } from 'react';
import { logger } from '@/lib/logger';

interface ErrorBoundaryState {
  hasError: boolean;
  error?: Error;
  errorInfo?: string;
}

interface ErrorBoundaryProps {
  children: ReactNode;
  fallback?: ReactNode;
  onError?: (error: Error, errorInfo: string) => void;
}

export class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundaryState> {
  constructor(props: ErrorBoundaryProps) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error: Error): ErrorBoundaryState {
    // Update state so the next render will show the fallback UI
    return {
      hasError: true,
      error,
    };
  }

  componentDidCatch(error: Error, errorInfo: React.ErrorInfo) {
    // Log the error with structured logging
    logger.error('React component error caught by boundary', {
      component: this.constructor.name,
      stack: error.stack,
      componentStack: errorInfo.componentStack,
      errorBoundary: true
    }, error);

    // Call optional error handler
    if (this.props.onError) {
      this.props.onError(error, errorInfo.componentStack);
    }

    this.setState({
      errorInfo: errorInfo.componentStack,
    });
  }

  private handleRetry = () => {
    this.setState({ hasError: false, error: undefined, errorInfo: undefined });
  };

  private handleReload = () => {
    window.location.reload();
  };

  render() {
    if (this.state.hasError) {
      // Custom fallback UI
      if (this.props.fallback) {
        return this.props.fallback;
      }

      // Default error UI
      return (
        <div className="min-h-screen bg-background flex items-center justify-center p-4">
          <div className="max-w-md w-full bg-card rounded-lg border p-6 text-center">
            <div className="mb-4">
              <div className="w-12 h-12 bg-destructive/10 rounded-full flex items-center justify-center mx-auto mb-4">
                <svg
                  className="w-6 h-6 text-destructive"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                >
                  <path
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth={2}
                    d="M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-2.5L13.732 4c-.77-.833-1.964-.833-2.732 0L3.732 16c-.77.833.192 2.5 1.732 2.5z"
                  />
                </svg>
              </div>
              <h2 className="text-xl font-semibold text-foreground mb-2">
                Something went wrong
              </h2>
              <p className="text-muted-foreground text-sm mb-6">
                We're sorry, but there was an unexpected error. Your conversation data is safe.
              </p>
            </div>

            <div className="space-y-3">
              <button
                onClick={this.handleRetry}
                className="w-full bg-primary text-primary-foreground hover:bg-primary/90 px-4 py-2 rounded-md text-sm font-medium transition-colors"
              >
                Try Again
              </button>
              <button
                onClick={this.handleReload}
                className="w-full bg-secondary text-secondary-foreground hover:bg-secondary/80 px-4 py-2 rounded-md text-sm font-medium transition-colors"
              >
                Reload Page
              </button>
            </div>

            {process.env.NODE_ENV === 'development' && this.state.error && (
              <details className="mt-6 text-left">
                <summary className="cursor-pointer text-sm text-muted-foreground mb-2">
                  Error Details (Development Only)
                </summary>
                <pre className="text-xs bg-muted p-3 rounded overflow-auto max-h-32">
                  {this.state.error.toString()}
                  {this.state.errorInfo}
                </pre>
              </details>
            )}
          </div>
        </div>
      );
    }

    return this.props.children;
  }
}

// Specialized error boundaries for different parts of the app

export function ChatErrorBoundary({ children }: { children: ReactNode }) {
  return (
    <ErrorBoundary
      fallback={
        <div className="flex-1 flex items-center justify-center p-4">
          <div className="text-center">
            <div className="w-12 h-12 bg-destructive/10 rounded-full flex items-center justify-center mx-auto mb-4">
              <svg className="w-6 h-6 text-destructive" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M8 12h.01M12 12h.01M16 12h.01M21 12c0 4.418-3.582 8-8 8a8.955 8.955 0 01-4.906-1.469L3 21l2.469-5.094A8.955 8.955 0 013 12c0-4.418 3.582-8 8-8s8 3.582 8 8z" />
              </svg>
            </div>
            <h3 className="text-lg font-semibold mb-2">Chat Error</h3>
            <p className="text-muted-foreground text-sm mb-4">
              There was an issue with the chat interface. Please refresh to continue.
            </p>
            <button
              onClick={() => window.location.reload()}
              className="bg-primary text-primary-foreground px-4 py-2 rounded-md text-sm"
            >
              Refresh Chat
            </button>
          </div>
        </div>
      }
      onError={(error) => {
        logger.error('Chat component error', {
          component: 'ChatErrorBoundary',
          userAction: 'chat_interaction'
        }, error);
      }}
    >
      {children}
    </ErrorBoundary>
  );
}

export function SettingsErrorBoundary({ children }: { children: ReactNode }) {
  return (
    <ErrorBoundary
      fallback={
        <div className="p-4 bg-card rounded-lg border">
          <div className="text-center">
            <h3 className="text-sm font-medium mb-2">Settings Error</h3>
            <p className="text-xs text-muted-foreground mb-3">
              Unable to load settings. Using defaults.
            </p>
            <button
              onClick={() => window.location.reload()}
              className="text-xs bg-secondary px-3 py-1 rounded"
            >
              Reload
            </button>
          </div>
        </div>
      }
      onError={(error) => {
        logger.error('Settings component error', {
          component: 'SettingsErrorBoundary',
          userAction: 'settings_interaction'
        }, error);
      }}
    >
      {children}
    </ErrorBoundary>
  );
}