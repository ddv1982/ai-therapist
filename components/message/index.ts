/**
 * Message Component System - Clean Public API
 */

// Main components
export { Message } from './Message';
export { MessageAvatar } from './MessageAvatar';
export { MessageContent } from './MessageContent';
export { MessageTimestamp } from './MessageTimestamp';

// Types
export type { MessageData, MessageProps } from './Message';

// Re-export design system types for convenience
export type { MessageRole } from '@/lib/design-system/message';