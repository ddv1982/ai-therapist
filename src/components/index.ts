/**
 * Components Domain - Master Exports
 * Centralized exports for all component domains
 */

// Domain-specific exports
export * as auth from './auth';
export * as chat from './chat';
export * as therapy from './therapy';
export * as messages from './messages';
export * as ui from './ui';
export * as providers from './providers';

// Direct exports for commonly used components
export * from './messages';
export * from './providers';

// UI components (flattened for convenience)
export * from './ui/primitives';
export * from './ui/enhanced';
export * from './ui/layout';

// Backward compatible exports for easier migration
// Auth components
export * from './auth/auth-guard';
export * from './auth/security-settings';

// Chat components
export * from './chat/chat-interface';
export * from './chat/chat-settings';
export * from './chat/session-controls';
export * from './chat/session-sidebar';

// Message components
export * from './messages/message';
export * from './messages/message-avatar';
export * from './messages/message-content';
export * from './messages/message-timestamp';

// Provider components
export * from './providers/theme-provider';

// Layout exports for backward compatibility
export { ErrorBoundary } from './ui/layout/error-boundary';