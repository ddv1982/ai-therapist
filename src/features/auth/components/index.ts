/**
 * Authentication Components Domain
 * Components for authentication and user management
 */

export * from './auth-guard';