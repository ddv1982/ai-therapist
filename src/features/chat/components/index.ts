/**
 * Chat Components Domain
 * Components for chat interface, sessions, and messaging functionality
 */

export * from './chat-settings';
export * from './session-controls';
export * from './session-sidebar';
export * from './typing-indicator';
export * from './virtualized-message-list';