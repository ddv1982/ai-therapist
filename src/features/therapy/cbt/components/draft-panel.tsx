import React from 'react';
import { Button } from '@/components/ui/button';
import { Trash2, Brain, Sparkles } from 'lucide-react';
import {useTranslations} from 'next-intl';

interface DraftPanelProps {
  hasDraft: boolean;
  draftLastSaved?: string;
  onDeleteDraft: () => void;
  onResume: () => void;
  onStartFresh: () => void;
}

export function DraftPanel({ hasDraft, draftLastSaved, onDeleteDraft, onResume, onStartFresh }: DraftPanelProps) {
  const t = useTranslations('cbt');

  if (!hasDraft) {
    return (
      <div className="flex justify-center">
        <Button
          onClick={onResume}
          className="justify-center gap-3 h-12 rounded-xl shadow-lg hover:shadow-xl transition-all duration-200 group px-8"
        >
          <div className="h-6 w-6 rounded-full bg-white/20 flex items-center justify-center group-hover:bg-white/30 transition-colors">
            <Brain className="w-4 h-4" />
          </div>
          <span className="font-semibold">{t('actions.begin')}</span>
          <Sparkles className="w-4 h-4 opacity-60 group-hover:opacity-100 transition-opacity" />
        </Button>
      </div>
    );
  }

  return (
    <div className="space-y-4">
      <div className="p-4 rounded-lg bg-blue-50 dark:bg-blue-950/30 border border-blue-200 dark:border-blue-800 text-blue-700 dark:text-blue-300">
        <div className="flex items-center justify-between mb-2">
          <div className="flex items-center gap-2">
            <h3 className="font-semibold">üìù Previous Session Found</h3>
            {hasDraft && (
              <span className="px-2 py-1 text-sm bg-green-100 dark:bg-green-900/30 text-green-700 dark:text-green-400 rounded-full">
                {t('status.saved')}
              </span>
            )}
          </div>
          <Button
            variant="ghost"
            size="icon"
            aria-label={t('actions.deleteDraft')}
            title={t('actions.deleteDraft')}
            onClick={onDeleteDraft}
          >
            <Trash2 className="w-4 h-4" />
          </Button>
        </div>
        <p className="text-sm opacity-90">
          {t('draft.found', { date: draftLastSaved ? new Date(draftLastSaved).toLocaleDateString() : 'recently' })}
        </p>
        <p className="text-sm opacity-75 mt-1">
          {t('draft.choice')}
        </p>
      </div>

      <div className="flex flex-col sm:flex-row gap-3 justify-center">
        <Button
          onClick={onResume}
          className="justify-center gap-3 h-12 rounded-xl shadow-lg hover:shadow-xl transition-all duration-200 group px-6"
        >
          <div className="h-6 w-6 rounded-full bg-white/20 flex items-center justify-center group-hover:bg-white/30 transition-colors">
            <Brain className="w-4 h-4" />
          </div>
          <span className="font-semibold">{t('actions.resume')}</span>
          <Sparkles className="w-4 h-4 opacity-60 group-hover:opacity-100 transition-opacity" />
        </Button>
        <Button
          onClick={onStartFresh}
          variant="outline"
          className="justify-center h-12 rounded-xl border-2 hover:bg-accent hover:text-accent-foreground px-6 font-semibold"
        >
          {t('actions.startNew')}
        </Button>
      </div>
    </div>
  );
}
