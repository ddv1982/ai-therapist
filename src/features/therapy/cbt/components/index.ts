export { SituationSection } from './situation-section';
export { EmotionsSection } from './emotions-section';
export { ThoughtsSection } from './thoughts-section';
export { SchemaSection } from './schema-section';
export { ChallengeSection } from './challenge-section';
export { ResultsSection } from './results-section';
export { EmotionScale } from './emotion-scale';
export { ArrayField } from './array-field';