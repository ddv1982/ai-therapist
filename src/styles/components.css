@layer components {
  /* Spacing utilities */
  .spacing-therapy-xs { @apply p-therapy-xs; }
  .spacing-therapy-sm { @apply p-therapy-sm; }
  .spacing-therapy-md { @apply p-therapy-md; }
  .spacing-therapy-lg { @apply p-therapy-lg; }
  .spacing-therapy-xl { @apply p-therapy-xl; }
  .spacing-therapy-2xl { @apply p-therapy-2xl; }

  /* Color variants */
  .therapy-primary { @apply bg-primary text-primary-foreground shadow-sm; }
  .therapy-secondary { @apply bg-secondary text-secondary-foreground; }
  .therapy-accent { @apply bg-accent text-accent-foreground shadow-sm; }
  .therapy-muted { @apply bg-muted text-muted-foreground; }
  .therapy-card { @apply bg-card text-card-foreground border border-border rounded-therapy-md shadow-sm; }

  /* Buttons */
  .therapy-button-primary { 
    @apply inline-flex items-center justify-center px-therapy-md py-therapy-xs rounded-therapy-sm text-base font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none bg-primary text-primary-foreground shadow-sm hover:bg-primary/90 active:scale-95; 
  }
  .therapy-button-secondary { 
    @apply inline-flex items-center justify-center px-therapy-md py-therapy-xs rounded-therapy-sm text-base font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none bg-secondary text-secondary-foreground hover:bg-secondary/80 active:scale-95; 
  }
  .therapy-button-accent { 
    @apply inline-flex items-center justify-center px-therapy-md py-therapy-xs rounded-therapy-sm text-base font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none bg-accent text-accent-foreground shadow-sm hover:bg-accent/90 active:scale-95; 
  }
  .therapy-button-ghost { 
    @apply inline-flex items-center justify-center px-therapy-md py-therapy-xs rounded-therapy-sm text-base font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none hover:bg-accent/10 hover:text-accent-foreground active:scale-95; 
  }

  /* Forms */
  .therapy-form-group { @apply space-y-therapy-xs; }
  .therapy-form-label { @apply text-sm font-medium text-foreground; }
  .therapy-form-description { @apply text-xs text-muted-foreground; }
  .therapy-form-error { @apply text-xs text-destructive; }
  .therapy-input { @apply w-full px-therapy-sm py-therapy-xs bg-background border border-input rounded-therapy-sm text-base focus:outline-none focus:ring-2 focus:ring-ring focus:border-transparent placeholder:text-muted-foreground disabled:opacity-50 disabled:cursor-not-allowed transition-colors duration-200; }

  /* Modals */
  .therapy-modal { @apply bg-background border border-border rounded-therapy-lg shadow-xl backdrop-blur-sm; }
  .therapy-modal-content { @apply p-therapy-md max-h-[90vh] overflow-y-auto; }
  .therapy-modal-header { @apply p-therapy-md border-b border-border flex items-center justify-between; }
  .therapy-modal-footer { @apply p-therapy-md border-t border-border flex items-center justify-end gap-therapy-sm; }

  /* Chat-specific */
  .message-spacing { @apply mb-4; }
  .message-padding { @apply p-2; }
  .message-content-user, .message-content-assistant { @apply leading-relaxed; }

  /* Glassmorphism */
  .glass-card { @apply bg-card/80 backdrop-blur-md border border-border/50 shadow-lg; }

  /* Animations */
  .animate-message-in { animation: fade-in 0.5s ease-out; }
  .animate-slide-in { animation: slide-in 0.3s ease-out; }

  /* Scrollbars */
  .custom-scrollbar::-webkit-scrollbar { width: 0; height: 0; }
  .custom-scrollbar { scrollbar-width: none; }
  .custom-scrollbar::-webkit-scrollbar-thumb { @apply bg-muted-foreground/20 rounded-full; }

  /* Tables */
  .therapeutic-table { @apply w-full border-primary/20 bg-primary/5 rounded-lg overflow-hidden; table-layout: fixed; }
  .table-container { @apply rounded-md border overflow-hidden mb-4; }
  .table-system { @apply w-full border-collapse; }
  .table-header-therapeutic { @apply bg-primary/10 text-primary-foreground/90 font-semibold py-3 px-4; }
  .table-row-therapeutic { @apply hover:bg-primary/10 transition-colors border-b border-primary/10; }
  .table-cell-therapeutic { @apply py-3 px-4 text-left; }

  /* Cards */
  .therapeutic-card { @apply transition-all duration-200 ease-out relative overflow-hidden; }
  .therapeutic-card:hover { @apply -translate-y-1; }
  .therapeutic-card::before { @apply absolute inset-0 bg-gradient-to-br from-primary/5 via-transparent to-accent/5 opacity-0 transition-opacity duration-300 pointer-events-none; content: ''; }
  .therapeutic-card:hover::before { @apply opacity-100; }

  /* Sliders */
  input[type="range"] { -webkit-appearance: none; appearance: none; background: transparent; cursor: pointer; }
  input[type="range"]::-webkit-slider-track { @apply bg-muted h-2 rounded-lg; }
  input[type="range"]::-webkit-slider-thumb { -webkit-appearance: none; @apply w-5 h-5 rounded-full bg-primary shadow-lg cursor-pointer transition-all; }
  input[type="range"]::-webkit-slider-thumb:hover { @apply scale-110 shadow-xl; }
}
