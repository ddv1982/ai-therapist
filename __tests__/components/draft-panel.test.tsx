import React from 'react';
import { render } from '@testing-library/react';
import { NextIntlClientProvider } from 'next-intl';
import { DraftPanel } from '@/features/therapy/cbt/components/draft-panel';
import enMessages from '@/i18n/messages/en.json';

function renderWithIntl(ui: React.ReactNode) {
  return render(
    <NextIntlClientProvider locale="en" messages={enMessages}>
      {ui}
    </NextIntlClientProvider>
  );
}

describe('DraftPanel', () => {
  it('renders with draft found', () => {
    const { container } = renderWithIntl(
      <DraftPanel
        hasDraft
        draftLastSaved="2025-10-19T00:00:00.000Z"
        onDeleteDraft={() => {}}
        onResume={() => {}}
        onStartFresh={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders without draft (begin button)', () => {
    const { container } = renderWithIntl(
      <DraftPanel hasDraft={false} onDeleteDraft={() => {}} onResume={() => {}} onStartFresh={() => {}} />
    );
    expect(container).toMatchSnapshot();
  });
});

