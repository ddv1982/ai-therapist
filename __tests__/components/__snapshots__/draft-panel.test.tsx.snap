// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`DraftPanel renders with draft found 1`] = `
<div>
  <div
    class="space-y-4"
  >
    <div
      class="p-4 rounded-lg bg-blue-50 dark:bg-blue-950/30 border border-blue-200 dark:border-blue-800 text-blue-700 dark:text-blue-300"
    >
      <div
        class="flex items-center justify-between mb-2"
      >
        <div
          class="flex items-center gap-2"
        >
          <h3
            class="font-semibold"
          >
            üìù Previous Session Found
          </h3>
          <span
            class="px-2 py-1 text-sm bg-green-100 dark:bg-green-900/30 text-green-700 dark:text-green-400 rounded-full"
          >
            Saved
          </span>
        </div>
        <button
          aria-label="Delete draft"
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-base font-semibold transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-muted hover:text-foreground h-12 w-12"
          title="Delete draft"
        >
          <div
            class="w-4 h-4"
            data-testid="trash-icon"
          />
        </button>
      </div>
      <p
        class="text-sm opacity-90"
      >
        We found an in-progress CBT session.
      </p>
      <p
        class="text-sm opacity-75 mt-1"
      >
        Choose to continue where you left off or start fresh.
      </p>
    </div>
    <div
      class="flex flex-col sm:flex-row gap-3 justify-center"
    >
      <button
        class="inline-flex items-center whitespace-nowrap text-base font-semibold focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-gradient-to-r from-primary to-accent text-primary-foreground hover:from-primary/90 hover:to-accent/90 active:from-primary/80 active:to-accent/80 py-2 justify-center gap-3 h-12 rounded-xl shadow-lg hover:shadow-xl transition-all duration-200 group px-6"
      >
        <div
          class="h-6 w-6 rounded-full bg-white/20 flex items-center justify-center group-hover:bg-white/30 transition-colors"
        >
          <div
            class="w-4 h-4"
            data-testid="brain-icon"
          />
        </div>
        <span
          class="font-semibold"
        >
          Resume Previous Session
        </span>
        <div
          class="w-4 h-4 opacity-60 group-hover:opacity-100 transition-opacity"
          data-testid="sparkles-icon"
        />
      </button>
      <button
        class="inline-flex items-center gap-2 whitespace-nowrap text-base transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border-input bg-background shadow-sm py-2 justify-center h-12 rounded-xl border-2 hover:bg-accent hover:text-accent-foreground px-6 font-semibold"
      >
        Start New Session
      </button>
    </div>
  </div>
</div>
`;

exports[`DraftPanel renders without draft (begin button) 1`] = `
<div>
  <div
    class="flex justify-center"
  >
    <button
      class="inline-flex items-center whitespace-nowrap text-base font-semibold focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-gradient-to-r from-primary to-accent text-primary-foreground hover:from-primary/90 hover:to-accent/90 active:from-primary/80 active:to-accent/80 py-2 justify-center gap-3 h-12 rounded-xl shadow-lg hover:shadow-xl transition-all duration-200 group px-8"
    >
      <div
        class="h-6 w-6 rounded-full bg-white/20 flex items-center justify-center group-hover:bg-white/30 transition-colors"
      >
        <div
          class="w-4 h-4"
          data-testid="brain-icon"
        />
      </div>
      <span
        class="font-semibold"
      >
        Begin CBT Session
      </span>
      <div
        class="w-4 h-4 opacity-60 group-hover:opacity-100 transition-opacity"
        data-testid="sparkles-icon"
      />
    </button>
  </div>
</div>
`;
